operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/informer/TemporaryResourceCache.java

```diff
@@ -38,8 +38,11 @@ public class TemporaryResourceCache<T extends HasMetadata> {
 
   private static final Logger log = LoggerFactory.getLogger(TemporaryResourceCache.class);
   private static final int MAX_RESOURCE_VERSIONS = 256;
+  private static final int TOMBSTONE_RETENTION_TIME = 120000;
```

Comment:
Wondering about this value, if high enough.

Reply:
It is certianly possible there could be an extreme scenario where the client is configured for a bunch of exponential retries - I'm not sure there is a hard cap on the amount of time in that case. We can keep things for longer, or you can go back the other direction of telling the cache prior to the create that you are intending to create the given resource.